### PR TITLE
[WIP] Issue #1805 Install dep in Makefile

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
Tested in Fedora 27 and it works fine there.

**Comment by LalatenduMohanty** [comment]
Tested on macOS and it working fine.

**Comment by LalatenduMohanty** [comment]
> I am able to build from master it when manually downloading the dep version 0.3.2 in GOPATH/bin.

@budhrg Where did the dep got installed in Windows bydefault?

**Comment by LalatenduMohanty** [comment]
@budhrg I think the last three dots `...` might be causing issue in Windows along with need for `$(GOPATH)/bin/dep$(IS_EXE)` . Will check in Windows.

**Comment by LalatenduMohanty** [comment]
Also we need see if auto installation of this is causing any issue in CI runs as we have specific version pinned in CI runs for dep.

File: Makefile [hunk 1]
```diff
@@ -84,7 +84,7 @@ __check_defined = \
 .PHONY: $(GOPATH)/bin/minishift$(IS_EXE)
 $(GOPATH)/bin/minishift$(IS_EXE): $(ADDON_ASSET_FILE) vendor
 	go install -tags "$(BUILD_TAGS)" -pkgdir=$(ADDON_BINDATA_DIR) -ldflags="$(VERSION_VARIABLES)" ./cmd/minishift
-vendor:
+vendor: $(GOPATH)/bin/dep
 	dep ensure -v
```
Comment (budhrg) [inline Makefile]: @LalatenduMohanty Could you add Add `echo $(shell dep version)` here to see which version of dep is installed?

**Comment by LalatenduMohanty** [comment]
Looks like CI (e.g. appveyor) will start using the dep version from master as `go get` always installs from master.  We do not want to use  `dep` from master as it will break our CI when the master gets broken. For the same reason we have pinned version of `dep` in CI. Hence closing the PR.
```
echo dep:  version     : devel  build date  :   git hash    :   go version  : go1.8.3  go compiler : gc  platform    : windows/amd64
dep: version : devel build date : git hash : go version : go1.8.3 go compiler : gc platform : windows/amd64
```

**Comment by gbraad** [comment]
We do the same for go-bindata?!?

**Pull request closed without merging**